iris/test.cc

```diff
@@ -0,0 +1,197 @@
+                    },
+                    std::move(request));
+  });
+
+  EXPECT_EQ("Successfully imported 150 flowers.\n",
+            HTTP(POSTFromFile(Printf("http://localhost:%d/import", FLAGS_iris_port), "dataset.tsv", "text/tsv"))
+                .body);
+
+  // Ref.: http://localhost:3000/stream
+  api.ExposeViaHTTP(FLAGS_iris_port, "/stream");
+
+  // The very first flower.
+  const auto result1 = ParseJSON<std::unique_ptr<Padawan>>(
+      HTTP(GET(Printf("http://localhost:%d/stream?cap=1", FLAGS_iris_port))).body);
+  const LabeledFlower& flower1 = *static_cast<const LabeledFlower*>(result1.get());
+  EXPECT_DOUBLE_EQ(5.1, flower1.SL);
+  EXPECT_DOUBLE_EQ(3.5, flower1.SW);
+  EXPECT_DOUBLE_EQ(1.4, flower1.PL);
+  EXPECT_DOUBLE_EQ(0.2, flower1.PW);
+  EXPECT_EQ("setosa", flower1.label);
+
+  // The very last flower.
+  const auto result2 = ParseJSON<std::unique_ptr<Padawan>>(
+      HTTP(GET(Printf("http://localhost:%d/stream?n=1", FLAGS_iris_port))).body);
+  const LabeledFlower& flower2 = *static_cast<const LabeledFlower*>(result2.get());
+  EXPECT_DOUBLE_EQ(5.9, flower2.SL);
+  EXPECT_DOUBLE_EQ(3.0, flower2.SW);
+  EXPECT_DOUBLE_EQ(5.1, flower2.PL);
+  EXPECT_DOUBLE_EQ(1.8, flower2.PW);
+  EXPECT_EQ("virginica", flower2.label);
+
+  if (FLAGS_run) {
+    // Ref.: http://localhost:3000/get?id=42
+    HTTP(FLAGS_iris_port).Register("/get", [&api](Request request) {
+      const auto id = FromString<size_t>(request.url.query["id"]);
+      api.Transaction([id](TestAPI::T_DATA data) { return data[id]; }, std::move(request));
+    });
+
+    // Ref.: [POST] http://localhost:3000/add?label=setosa&sl=5&sw=5&pl=5&pw=5
+    HTTP(FLAGS_iris_port).Register("/add", [&api](Request request) {
+      const std::string label = request.url.query["label"];
+      const auto sl = FromString<double>(request.url.query["sl"]);
+      const auto sw = FromString<double>(request.url.query["sw"]);
+      const auto pl = FromString<double>(request.url.query["pl"]);
+      const auto pw = FromString<double>(request.url.query["pw"]);
+      // In real life this should be a POST.
+      if (!label.empty()) {
+        const LabeledFlower flower(++number_of_flowers, sl, sw, pl, pw, label);
+        api.Transaction([flower](TestAPI::T_DATA data) {
+                          data.Add(flower);
+                          return "OK\n";
+                        },
+                        std::move(request));
+      } else {
+        request("Need non-empty label, as well as sl/sw/pl/pw.\n");
+      }
+    });
+
+    // Ref.: http://localhost:3000/viz
+    // Ref.: http://localhost:3000/viz?x=1&y=2
+    HTTP(FLAGS_iris_port).Register("/viz", [&api](Request request) {
+      auto x_dim = std::min(size_t(3), FromString<size_t>(request.url.query.get("x", "0")));
+      auto y_dim = std::min(size_t(3), FromString<size_t>(request.url.query.get("y", "1")));
+      if (y_dim == x_dim) {
+        y_dim = (x_dim + 1) % 4;
+      }
+      struct PlotIrises {
```

Comment:
Reorder suggestion - move `struct PlotIrises` up, and the code about `x_dim/y_dim` down to the `api.Transaction(...)`. IMHO it would be more readable.